### PR TITLE
Add method for convert definitions in to dict

### PR DESCRIPTION
**Pull request by filimarc**
Add a _to_dict() method for Definition to convert to dict. Add test for this function in test_neuron_models.py

### CODE REVIEW

**Review by drodarie — APPROVED** [review]
LGTM

**Review by Helveg — CHANGES_REQUESTED** [review]
Dataclasses can already do this, please refactor to make use of https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict

File: tests/test_neuron_models.py [hunk 1]
```diff
@@ -122,3 +122,66 @@ def test_morphology(self):
         n_locs = sum(len(c.points) for c in self.p75_pas.cables)
         self.assertEqual(len(self.p75_pas.cables), len(cell.sections), "missing cables")
         self.assertEqual(n_locs, sum(s.n3d() for s in cell.sections), "missing locs")
+
+
+@unittest.skipIf(
+    "NRN_SEGFAULT" in os.environ,
+    "These tests are skipped to test the other tests below separately. See https://github.com/neuronsimulator/nrn/issues/2641",
+)
```
Comment (Helveg) [inline tests/test_neuron_models.py:130]: Has this been confirmed as required for this test suite?

Comment (filimarc) [reply tests/test_neuron_models.py:130]: Actually, this is not necessary. I should remove it.

**Review by Helveg — APPROVED** [review]

**Pull request merged**